### PR TITLE
feat: use new build script and kmc

### LINKED ISSUE #224: feat: build with kmc instead of kmlmc
Part of the 17.0-alpha work. @mcdurdin will work on this.

TODO:
- [x] We need to verify against the older kmlmp output for packages.
- [x] Add LICENSE.md to each package (#225)
- [x] Create a filelist and build it with a single command for performance (#226)

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #224.

Replaces the old build script with a cleaner version that utilizes builder.inc.sh, standardizing our build commands.

Uses `kmc` to build all source models, in a single command rather than one-by-one as previously. Handles external models more simply.

TODO: 
- [x] Awaiting merge of changes to kmc to keymanapp/keyman/master before committing updated package.json; this will not build until the updated kmc is available. (See keymanapp/keyman#9717)

---

Please note: this will merge into staging-17.0, in preparation for 17.0 release. Changes on the staging-17.0 branch will not be published to keyman.com, until they are merged into master when we go to release (or when we determine the updated build tools are sufficiently stable).

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -1,105 +1,147 @@
...
+if [[ -z ${KMC+x} ]]; then
+  export KMC="${REPO_ROOT}/node_modules/.bin/kmc"
+  readonly KMC
 fi
 
-#
-# Collect filenames
-#
+# TODO: remove -W
+export KMC_BUILD_PARAMS="build $builder_debug --for-publishing"
```
Comment (darcywong00) [inline build.sh:46]: Should `--for-publishing` be a separate builder publish action?

File: resources/build_targets.inc.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#
+# Build a list of all targets that we will be building, in a specific order:
+# * release
+# * experimental
```
Comment (darcywong00) [inline resources/build_targets.inc.sh:4]: Doesn't this also add to the list of external targets in build_external_targets.txt?

File: build.sh [hunk 3]
```diff
@@ -1,105 +1,147 @@
...
+. "${THIS_SCRIPT%/*}/resources/builder.inc.sh"
+# END STANDARD BUILD SCRIPT INCLUDE
 
-# Master json schema is from https://api.keyman.com/schemas/model_info.json
+# script runs from repo root
+cd "$THIS_SCRIPT_PATH"
 
-MODELINFO_SCHEMA_JSON="$MODELROOT/tools/model_info.source.json"
-MODELINFO_SCHEMA_DIST_JSON="$MODELROOT/tools/model_info.distribution.json"
+. "./resources/util.inc.sh"
```
Comment (darcywong00) [inline build.sh:17]: Is this still needed?
I thought `die` can be replaced with `builder_die` (external.inc.sh). And I didn't see `parse_args` still being called.

File: build.sh (hunk 1 above)
Comment (mcdurdin) [reply build.sh:46]: No, because this is a flag to tell kmc to build the .model_info files.

File: resources/build_targets.inc.sh (hunk 2 above)
Comment (mcdurdin) [reply resources/build_targets.inc.sh:4]: I think so, yes.

File: build.sh (hunk 3 above)
Comment (mcdurdin) [reply build.sh:17]: Will review

Comment (mcdurdin) [reply build.sh:17]: ci.sh is still using a lot of the functions in util.inc.sh, so that will need separate cleanup.

**Comment by darcywong00** [comment]
Should ./build.sh of the repo work on Linux now? I got these warnings/errors
```
gonzalez_quint_coto.cjp-latn.cabecar.model.kps - warn KM0400F: The file ..\build\gonzalez_quint_coto.cjp-latn.cabecar.model.js does not follow the recommended filename conventions. The extension should be all lower case, and the filename should include only alphanumeric characters, -, _, + and .
gonzalez_quint_coto.cjp-latn.cabecar.model.kps - warn KM0400F: The file ..\LICENSE.md does not follow the recommended filename conventions. The extension should be all lower case, and the filename should include only alphanumeric characters, -, _, + and .
gonzalez_quint_coto.cjp-latn.cabecar.model.kps - info KM05007: release/gonzalez_quint_coto/gonzalez_quint_coto.cjp-latn.cabecar/source/gonzalez_quint_coto.cjp-latn.cabecar.model.kps failed to build.
gonzalez_quint_coto.cjp-latn.cabecar.kpj - info KM05010: The build failed because option "treat warnings as errors" is enabled and there are one or more warnings.
gonzalez_quint_coto.cjp-latn.cabecar.kpj - info KM0500C: Project release/gonzalez_quint_coto/gonzalez_quint_coto.cjp-latn.cabecar/gonzalez_quint_coto.cjp-latn.cabecar.kpj failed to build.

```

Is it not liking the path separator `\`?

**Comment by mcdurdin** [comment]
> Should ./build.sh of the repo work on Linux now? I got these warnings/errors

These look like bugs in kmc-package which we recently fixed in [keymanapp/keyman#](https://github.com/keymanapp/keyman/pull/10064). So we should re-test with that version on Linux before merging this.

**Comment by mcdurdin** [comment]
Tested with 17.0.222-alpha and this has now been resolved, looking for a review thx

File: package-lock.json [hunk 4]
```diff
@@ -1,78 +1,250 @@
...
   "packages": {
     "": {
       "name": "lexical-models",
       "version": "1.0.0",
       "license": "MIT",
       "dependencies": {
-        "@keymanapp/lexical-model-compiler": "^16.0.138",
-        "@keymanapp/models-types": "^16.0.138",
+        "@keymanapp/kmc": "17.0.192-alpha",
+        "@keymanapp/models-types": "17.0.192-alpha",
```
Comment (darcywong00) [inline package-lock.json:13]: My package-lock.json got updated to 17.0.222-alpha (numerous lines)

Comment (mcdurdin) [reply package-lock.json]: yep, `npm install` once more sorts that. First time did not update the dependency versions, is it a bug in npm? No idea...

**Review by darcywong00 — APPROVED** [review]
lgtm

**Pull request merged**